src/include/sof/audio/module_adapter/module/generic.h

```diff
@@ -141,12 +144,13 @@ struct module_processing_data {
 	void *out_buff; /**< A pointer to module output buffer. */
 };
 
+#ifndef SOF_MONOLITHIC_BUILD
 /*
  * Definition used to extend structure definitions to include fields for exclusive use by SOF.
  * This is a temporary solution used until work on separating a common interface for loadable
  * modules is completed.
  */
-#define SOF_MODULE_API_PRIVATE
+#define SOF_MONOLITHIC_BUILD
```

Comment:
no need to put this under `#ifndef SOF_MONOLITHIC_BUILD` and it makes little sense that way - if it isn't defined, then we define it... From this commit I don't see any other places where this can be defined, so this doesn't seem to protect against redefinition.